### PR TITLE
Allow ports to be optional when adding servers

### PR DESCRIPTION
**Pull request by grubernaut**
When updating a clients servers, as nomad does not use the gossip
protocol over a specified port for clients, it was required to specify
ports along with server addresses.

Now specifying ports are optional, and if unspecified the default `4647`
port is used, reflecting a notice back to the user.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -324,7 +325,26 @@ func (c *Client) SetServers(servers []string) {
...
+			if strings.Contains(err.Error(), "missing port") {
+				c.logger.Printf("[WARN] client: port not specified, using default port")
+				setServers[i] = fmt.Sprintf("%s:4647", setServers[i])
+			} else {
+				c.logger.Printf("[ERR] client: error in server address %s", err)
+			}
+		}
+	}
+
+	c.logger.Printf("[INFO] client: adding servers '%s'", setServers)
```
Comment (dadgar) [inline client/client.go]: adding servers: %s (or %v)

File: client/client.go [hunk 2]
```diff
@@ -324,7 +325,26 @@ func (c *Client) SetServers(servers []string) {
...
+	setServers := make([]string, len(servers))
+	copy(setServers, servers)
+	for i := 0; i < len(setServers); i++ {
+		if _, _, err := net.SplitHostPort(setServers[i]); err != nil {
+			// multiple errors can be returned here, only searching for mising
+			if strings.Contains(err.Error(), "missing port") {
+				c.logger.Printf("[WARN] client: port not specified, using default port")
+				setServers[i] = fmt.Sprintf("%s:4647", setServers[i])
+			} else {
+				c.logger.Printf("[ERR] client: error in server address %s", err)
```
Comment (dadgar) [inline client/client.go]: `c.logger.Printf("[WARN] client: server address %v invalid: %v", setServers[i], err)`

**Comment by voronshki-vallavitch** [comment]
@grubernaut HI SWEETIE, PLEASE COME TO OUTER YALLAVILLE, WE LOVE YOU,, YOUR CODE IS SO BEAUTIFUL,, SO MUCH LOVE FOR YOU

**Comment by voronshki-vallavitch** [comment]
@grubernaut HI SWEETIE, PLEASE COME TO OUTER YALLAVILLE, WE LOVE YOU,, YOUR CODE IS SO BEAUTIFUL, SO MUCH LOVE

File: client/client.go [hunk 3]
```diff
@@ -324,7 +325,26 @@ func (c *Client) SetServers(servers []string) {
...
-	c.servers = servers
+	// net.ResolveTCPAddr requires port to be set, if one is not provided, supply default port
+	// Using net.SplitHostPort in the event of IPv6 addresses with multiple colons.
+	// IPv6 addresses must be passed in with brackets,
+	// i.e: [::1]:4647 or [::1]
+	setServers := make([]string, len(servers))
+	copy(setServers, servers)
+	for i := 0; i < len(setServers); i++ {
+		if _, _, err := net.SplitHostPort(setServers[i]); err != nil {
+			// multiple errors can be returned here, only searching for mising
```
Comment (ryanuber) [inline client/client.go]: "missing"

File: client/client.go [hunk 4]
```diff
@@ -324,7 +325,26 @@ func (c *Client) SetServers(servers []string) {
...
+	// IPv6 addresses must be passed in with brackets,
+	// i.e: [::1]:4647 or [::1]
+	setServers := make([]string, len(servers))
+	copy(setServers, servers)
+	for i := 0; i < len(setServers); i++ {
+		if _, _, err := net.SplitHostPort(setServers[i]); err != nil {
+			// multiple errors can be returned here, only searching for missing
+			if strings.Contains(err.Error(), "missing port") {
+				c.logger.Printf("[WARN] client: port not specified, using default port")
+				setServers[i] = fmt.Sprintf("%s:4647", setServers[i])
```
Comment (ryanuber) [inline client/client.go]: I'd probably just use `net.JoinHostPort` here, then I think we can omit `strings` too

File: client/client.go [hunk 5]
```diff
@@ -324,7 +325,26 @@ func (c *Client) SetServers(servers []string) {
...
+			} else {
+				c.logger.Printf("[WARN] client: server address %q invalid: %v", setServers[i], err)
+			}
+		}
+	}
+
+	c.logger.Printf("[INFO] client: adding servers: %s", setServers)
```
Comment (ryanuber) [inline client/client.go]: Maybe switch this like `setting server address list: %s` since we are swapping the whole thing out

**Comment by ryanuber** [comment]
Only minor feedback but LGTM

**Pull request merged**